components/google-cloud/google_cloud_pipeline_components/experimental/remote/gcp_launcher/job_remote_runner.py

```diff
@@ -0,0 +1,151 @@
+
+_POLLING_INTERVAL_IN_SECONDS = 20
+_CONNECTION_ERROR_RETRY_LIMIT = 5
+
+_JOB_COMPLETE_STATES = (
+    gca_job_state.JobState.JOB_STATE_SUCCEEDED,
+    gca_job_state.JobState.JOB_STATE_FAILED,
+    gca_job_state.JobState.JOB_STATE_CANCELLED,
+    gca_job_state.JobState.JOB_STATE_PAUSED,
+)
+
+_JOB_ERROR_STATES = (
+    gca_job_state.JobState.JOB_STATE_FAILED,
+    gca_job_state.JobState.JOB_STATE_CANCELLED,
+    gca_job_state.JobState.JOB_STATE_PAUSED,
+)
+
+
+class JobRemoteRunner():
+  """Common module for creating and poll jobs on the Vertex Platform."""
+
+  def __init__(self, job_type, project, location, gcp_resources):
+    """Initlizes a job client and other common attributes."""
+    self.job_type = job_type
+    self.project = project
+    self.location = location
+    self.gcp_resources = gcp_resources
+    self.client_options = {
+        'api_endpoint': location + '-aiplatform.googleapis.com'
+    }
+    self.client_info = gapic_v1.client_info.ClientInfo(
+        user_agent='google-cloud-pipeline-components',)
+    self.job_uri_prefix = f"https://{self.client_options['api_endpoint']}/v1/"
+    self.job_client = self.create_job_client(
+        client_options=self.client_options, client_info=self.client_info)
+
+  def create_job_client(self, client_options,
+                        client_info) -> aiplatform.gapic.JobServiceClient:
+    """Creates a job client using the google.cloud.aiplatform library."""
+    return aiplatform.gapic.JobServiceClient(
+        client_options=client_options, client_info=client_info)
+
+  def create_gcp_resource(self):
+    """Instantiate GCPResources Proto."""
+    return GcpResources().resources.add()
+
+  def check_if_job_exists(self, job_resource) -> Optional[str]:
+    """Check if the job already exists."""
+    if path.exists(
+        self.gcp_resources) and os.stat(self.gcp_resources).st_size != 0:
+      with open(self.gcp_resources) as f:
+        serialized_gcp_resources = f.read()
+        job_resource = json_format.Parse(serialized_gcp_resources, job_resource)
+        job_name = job_resource.resource_uri[len(self.job_uri_prefix):]
+
+        logging.info('%s name already exists: %s. Continue polling the status',
+                     self.job_type, job_name)
+      return job_name
+    else:
+      return None
+
+  def create_job(self, payload: str, job_resource) -> str:
+    """Create a job."""
+    parent = f'projects/{self.project}/locations/{self.location}'
+    # TODO(kevinbnaughton) remove empty fields from the spec temporarily.
+    job_spec = json.loads(payload, strict=False)
+    if self.job_type == 'CustomJob':
+      create_job_response = self.job_client.create_custom_job(
+          parent=parent, custom_job=job_spec)
+    elif self.job_type == 'BatchPredictionJob':
+      create_job_response = self.job_client.create_batch_prediction_job(
+          parent=parent, batch_prediction_job=job_spec)
```

Comment:
nit - it's better to pass the function as a parameter to avoid custom logic in this file. 

Also consider creating a constant value for 'CustomJob' and ' BatchPredictionJob' and reused in all places.

Reply:
Done, passing functions.